api/api.go

```diff
@@ -86,16 +86,33 @@ func (b *BlockChainAPI) BlockNumber() (hexutil.Uint64, error) {
 	return hexutil.Uint64(latestBlockHeight), nil
 }
 
-// Syncing returns false in case the node is currently not syncing with the network. It can be up-to-date or has not
-// yet received the latest block headers from its pears. In case it is synchronizing:
+// Syncing returns false in case the node is currently not syncing with the network.
+// It can be up-to-date or has not yet received the latest block headers from its peers.
+// In case it is synchronizing:
 // - startingBlock: block number this node started to synchronize from
 // - currentBlock:  block number this node is currently importing
 // - highestBlock:  block number of the highest block header this node has received from peers
-// - pulledStates:  number of state entries processed until now
-// - knownStates:   number of known state entries that still need to be pulled
 func (b *BlockChainAPI) Syncing() (interface{}, error) {
-	// todo maybe we should check if the node is caught up with the latest flow block
-	return false, nil
+	currentBlock, err := b.blocks.LatestCadenceHeight()
+	if err != nil {
+		return nil, err
+	}
+
+	highestBlock, err := b.evm.GetLatestBlockHeight(context.Background())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Add a todo to introduce caching if it will be needed. This feels to me it could be a request made a lot of times, but for sake of keeping it simple let's not assume on those things, but still add a todo so we don't forget to revisit.

Reply:
I guess the main usage of `eth_syncing` is to provide a real-time view of the indexing status. Introducing caching would defeat that purpose. I understand, however, the performance implications of multiple calls on this endpoint. Maybe the way to go about this is to provide some rate limiting, either on code level or infra level.

Reply:
Yeah I get that, what I'm curious to see is if this endpoint is being spammed by how clients are made, what I'm thinking is if there's sync requests done in an interval. If that will be true we could have caching that would have TTL so if multiple requests are made within a small time period you can assume we are not out of sync. Nevermind, I think I'm complicating